File: Data Structures & Algorithms/duplicate-integer/submission-6.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        map<int, int> dict;
        int len = nums.size();
        int curr_number;
        cout << len;
        for(int i=0;i<len;i++){
            curr_number = nums[i];
            if (dict.count(curr_number) == 0){
                dict[curr_number] = 1;
            }
            else{
                return true;
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/anagram-groups/submission-6.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        res = defaultdict(list)
        output = []
        for i in range(len(strs)):
            s = strs[i]
            count = [0] * 26
            for c in s:
                count[ord(c) - ord('a')] += 1    
            len_output = len(output)
            if res.get(tuple(count)) == None:
                res[tuple(count)] = len_output
                output.append([s])
            else:
                output[res[tuple(count)]].append(s)
        return output